crates/worker/src/partition_processor_manager/mod.rs

```diff
@@ -251,6 +271,9 @@ impl PartitionProcessorManager {
                 Some(partition_processor_rpc) = self.incoming_partition_processor_rpc.next() => {
                     self.on_partition_processor_rpc(partition_processor_rpc);
                 }
+                Some(result) = self.snapshot_export_tasks.next() => {
+                    self.on_snapshot_task_result(result.expect("export task ok??"));
+                }
                 _ = &mut shutdown => {
                     self.health_status.update(WorkerStatus::Unknown);
                     return Ok(());
```

Comment:
Once we break out of the loop here, we could stop the pending snapshot tasks via

```
for snapshot_task in self.snapshot_export_tasks.iter() {
            snapshot_task.abort();
        }

        Ok(())
```

Reply:
Done now!